test/integ/test_session_parameters.py

```diff
@@ -25,3 +25,72 @@ def test_session_parameters(db_parameters):
     ret = connection.cursor().execute(
         "show parameters like 'TIMEZONE'").fetchone()
     assert ret[1] == 'UTC'
+
+
+def test_client_session_keep_alive(db_parameters):
+    """Tests client_session_keep_alive setting.
+
+    Ensures that client's explicit config for client_session_keep_alive
+    session parameter is always honored and given higher precedence over
+    user and account level backend configuration.
+    """
+    # Ensure backend parameter is set to False
+    set_backend_client_session_keep_alive(db_parameters, False)
+    connection = create_client_connection(db_parameters, True)
```

Comment:
I think this extra function is not necessary. Try something like this instead: `with conn_cnx(client_session_keep_alive=True) as connection:`
You'll need to add `conn_cnx` argument to the test and indent the next 3 lines, but this will also get rid of the `close` call. That fixture inserts the necessary data like username and then puts the extra arguments you give it on top of the default ones. You can follow it here: https://github.com/snowflakedb/snowflake-connector-python/blob/2b41eaa9ce6a57a3a46ad683b9a89fc7ce408b8e/test/integ/conftest.py#L221

Reply:
I tried this and it didn't work. The parameter isn't being communicated it seems.